src/Algebra/Properties/Semiring.agda

```diff
@@ -0,0 +1,33 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some basic items and theory for Semiring.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Semiring)
+import Algebra.Properties.Monoid.Divisibility as MonoidDiv
+open import Data.Product using (_,_)
+
+module Algebra.Properties.Semiring {r₁ r₂} (R : Semiring r₁ r₂) where
+
+open Semiring R
+open MonoidDiv *-monoid using (_∣_; ∣-respˡ; ε∣_)
+
+------------------------------------------------------------------------------
+-- Divisibility properties specific to Semiring.
+
+1∣ : ∀ x → 1# ∣ x
+1∣ = ε∣_
+
+∣0 : ∀ x → x ∣ 0#
```

Comment:
This is `∣-max` from `Algebra.Divisibility`

Reply:
Do you suggest to replace
```
∣0 x = (0# , zeroˡ x)
```
with
```
∣0 x = ∣-max zeroˡ x
```
?
The former is easy to understand, and the latter is brain-twisting.
The reader would wonder at the proof: in what way the maximum notion is related to the fact that everything divides 0 ?
Also this requires to add
```
open import Algebra.Divisibility _≈_ _*_ using (∣-max)
```
To my mind, it is much better to tell "my cat eats everything" than
"my cat is maximal in the relation of eating". The reason is evident.

Well, the effect is weakened by that the reader still would pay attention only to the type 
``∣0 : ∀ x → x ∣ 0#``.

I set your version.
But I shall be glad if you return to the original version.

Reply:
You have a point. If you want you can remove `∣-max` and `∣-min` from `Algebra.Divisibility` and just write the proofs where needed.